reconcile/test/test_terraform_cloudflare_resources.py

```diff
@@ -0,0 +1,179 @@
+
+import reconcile.terraform_cloudflare_resources as integ
+
+
+@pytest.fixture
+def query_data(external_resources):
+    return TerraformCloudflareResourcesQueryData(
+        namespaces=[
+            NamespaceV1(name="namespace1", externalResources=[external_resources])
+        ],
+    )
+
+
+@pytest.fixture
+def settings():
+    return {}
+
+
+@pytest.fixture
+def provisioner_config():
+    return CloudflareAccountV1(
+        name="cfaccount",
+        providerVersion="3.22.0",
+        apiCredentials=VaultSecretV1(
+            path="",
+            field="",
+        ),
+        terraformStateAccount=AWSAccountV1(
+            name="awsaccount",
+            automationToken=VaultSecretV1(
+                path="",
+                field="",
+            ),
+            terraformState=TerraformStateAWSV1(
+                provider="s3",
+                bucket="bucket",
+                region="region",
+                integrations=[],
+            ),
+        ),
+    )
+
+
+@pytest.fixture
+def external_resources(provisioner_config):
+    return NamespaceTerraformProviderResourceCloudflareV1(
+        provider="cloudflare",
+        provisioner=provisioner_config,
+        resources=[
+            NamespaceTerraformResourceCloudflareZoneV1(
+                provider="zone",
+                zone="testzone.com",
+                plan="enterprise",
+                type="full",
+                settings='{"foo": "bar"}',
+                argo={
+                    "tiered_caching": True,
+                    "smart_routing": True,
+                },
+                records=[
+                    CloudflareZoneRecordV1(
+                        name="record",
+                        type="CNAME",
+                        ttl=5,
+                        value="example.com",
+                        proxied=False,
+                    ),
+                ],
+                workers=[
+                    CloudflareZoneWorkerV1(
+                        identifier="testworker",
+                        pattern="testzone.com/.*",
+                        script=CloudflareZoneWorkerScriptV1(
+                            name="testscript",
+                            content_from_github=CloudflareZoneWorkerScriptContentFromGithubV1(
+                                repo="foo",
+                                path="bar",
+                                ref="baz",
+                            ),
+                            vars=[
+                                CloudflareZoneWorkerScriptVarsV1(
+                                    name="somename",
+                                    text="sometext",
+                                )
+                            ],
+                        ),
+                    )
+                ],
+            ),
+        ],
+    )
+
+
+def test_build_specs(mocker, query_data):
+    get_github_file = mocker.patch.object(integ, "get_github_file")
```

Comment:
One thing I missed earlier was just adding `autospec=True` as a kwarg here so that the arguments passed to `get_github_file` are checked. Without this, we could pass in the wrong number of arguments and the test would still pass.

Reply:
this test has been removed with the removal of the `build_specs` method